static/app/views/dashboards/hooks/useDashboardsLimit.tsx

```diff
@@ -0,0 +1,93 @@
+import {tct} from 'sentry/locale';
+import {useApiQuery} from 'sentry/utils/queryClient';
+import useOrganization from 'sentry/utils/useOrganization';
+import type {DashboardListItem} from 'sentry/views/dashboards/types';
+
+// A minimal version of the Subscription type that we need for this hook
+interface Subscription {
+  planDetails: {
+    dashboardLimit: number;
+  };
+}
+
+interface UseDashboardsLimitResult {
+  dashboardsLimit: number;
+  hasReachedDashboardLimit: boolean;
+  isLoading: boolean;
+  limitMessage?: React.ReactNode;
+}
+
+const UNLIMITED_DASHBOARDS_LIMIT = -1;
+
+export function useDashboardsLimit(): UseDashboardsLimitResult {
+  const organization = useOrganization();
+  const {data: subscription, isLoading: isLoadingSubscription} =
+    useApiQuery<Subscription>([`/subscriptions/${organization.slug}/`], {
+      staleTime: Infinity,
+      enabled: organization.features.includes('dashboards-plan-limits'),
+    });
```

Comment:
Seems like there's an existing `useSubscription` hook, how come we don't use that? Unless it's fetching something different.

Reply:
Annoyingly, we can't import anything from `gsApp` (which is where that hook is) into these frontend files 😓 that's why I had to query for this manually